src/HIE/Bios/Ghc/Logger.hs

```diff
@@ -35,11 +31,11 @@ newtype LogRef = LogRef (IORef Builder)
 newLogRef :: IO LogRef
 newLogRef = LogRef <$> newIORef id
 
-readAndClearLogRef :: Options -> LogRef -> IO String
-readAndClearLogRef opt (LogRef ref) = do
+readAndClearLogRef :: LogRef -> IO String
+readAndClearLogRef (LogRef ref) = do
     b <- readIORef ref
     writeIORef ref id
-    return $! convert opt (b [])
+    return $! unlines (b [])
```

Comment:
Out of interest, I think this is one of the very few places where `($!)` is used. Is there a reason for it?

Reply:
This is old code from `hpp` still so I can't comment on the reason.